### DISCUSSION TITLE
Better Support with Get-PnpTenantSite for Sites With Associated Groups

### DISCUSSION THREAD

**Discussion by rednender** [pnp/powershell#191] [Ideas] [answered]
Currently, there is no easy way to get all sites that have an associated group.  You can't filter based on GroupId like you can with Get-SPOSite using -GroupIdDefined.  Also, using -Detailed doesn't return the associated GUID in either GroupId or RelatedGroupId (ruling out using a where clause).

The only solution I've found so far is: Get-PnPTenantSite | % {Get-PnPTenantSite -Url $_.Url}, which is very inefficient. 

Returning the GroupId or being able to filter based on GroupId being set would be extremely helpful.

**Comment by erwinvanhunen** [comment, marked as answer]
We just added it. It will be available in the 1.2.7-nightly release which will be available in a few hours from now. 

You can install the nightly release with

```powershell
Install-Module -Name PnP.PowerShell -AllowPrerelease -RequiredVersion 1.2.7-nightly
```

or update to the latest nightly release with

```powershell
Update-Module -Name PnP.PowerShell -AllowPrerelease -RequiredVersion 1.2.7-nightly
```
